### DISCUSSION TITLE
How to integrate Bevy App into an EGUI App ?

### DISCUSSION THREAD

**Discussion by Shrav108** [emilk/egui#5099] [Q&A]
I am trying to create an app which shows 3D Cad .STL files after some inputs from the user as per the given figure. I am using BEVY engine with ```bevy_stl``` to show the cad file.

![image](https://github.com/user-attachments/assets/788b384f-4de3-4534-b296-64760cef93f8)

How to include the bevy app into the egui app ?

**Comment by 9SMTM6** [comment]
As far as I understand you can embed egui in bevy. So, while it may be a bit more work on your side, I'd suggest going the other way around.

**Comment by gfxenjoyer** [comment]
You need to render your bevy app to a wgpu texture then pass the texture into your egui widget. There's an example of this in the `bevy_egui` crate. https://github.com/mvlabat/bevy_egui/blob/main/examples/render_to_image_widget.rs

The main issue is passing in user interaction from your egui widget into your bevy app. Iced has something similar to this called `custom_shader` with raw wgpu instead of bevy. https://github.com/iced-rs/iced/tree/master/examples/custom_shader
